app/services/json_schema/base.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require "json_schema/validation"
+require "json_schema/version_event_file_name"
+
+module JsonSchema
+  class Base
+    include InitializeWithConfig
+
+    def call(body:)
+      config[:schema] = schema
+      error_checker.call(config, body: body)
+    end
+
+  private
+
+    def schema
+      JSON.parse(File.read(json_schema_file_location.call(config)))
+    end
```

Comment:
Ehhh, this seems... convoluted. In `call` we change the `config` so that it includes `schema` field. and we get `schema` by calling some method to which we pass the old `config`? I sure hope that method doesn't need `schema` in it...

Can we minimise the size of config with which we call `json_schema_file_location`? A new hash would probably make it a lot easier to read.

Reply:
Yes. Making this an orchestration to pass in a schema as a param to call (and on down to error_checker), would make this cleaner. It strikes me that it's not as SOLID as it should be too and caching the schema is also something that may be required.

Reply:
Changed. This doesn't actually need to call the JSON parser. More solid if it is just passed the schema and the body to validate. Moved to lib as well.